src/main/java/org/dcsa/core/validator/RequiredIfTrue.java

```diff
@@ -0,0 +1,33 @@
+package org.dcsa.core.validator;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.*;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
+@Retention(RUNTIME)
+@Documented
+@Constraint(validatedBy = RequiredIfTrueValidator.class)
+public @interface RequiredIfTrue {
+
+    String isFieldReferenceRequired();
+
+    String fieldReference();
+
+    boolean allowNull() default false;

```

Comment:
I would remove this.  Where relevant, please can use `@NotNull` instead, which gives a better error message.  Replace it with `true` in the rest of the code will suffice.